### PR TITLE
add sql selects and migrations

### PR DESCRIPTION
**Pull request by dani3lz**

### CODE REVIEW

File: homework-selects/sql_questions.sql [hunk 1]
```diff
@@ -1,42 +1,197 @@
 -- Write a query to display: 
 -- 1. the first name, last name, salary, and job grade for all employees.
-SELECT first_name, last_name, salary, job_title
-FROM employees
-         LEFT JOIN jobs USING (job_id);
+SELECT FIRST_NAME, LAST_NAME, SALARY, JOB_TITLE
+FROM EMPLOYEES
+         LEFT JOIN JOBS USING (JOB_ID);
```
Comment (Evgenikaall) [inline homework-selects/sql_questions.sql:5]: I'll suggest you to write sql without specific features , for example using is feature of PL/SQL. Go with ON condition.

File: homework-selects/sql_questions.sql [hunk 2]
```diff
@@ -1,42 +1,197 @@
 -- Write a query to display: 
 -- 1. the first name, last name, salary, and job grade for all employees.
-SELECT first_name, last_name, salary, job_title
-FROM employees
-         LEFT JOIN jobs USING (job_id);
+SELECT FIRST_NAME, LAST_NAME, SALARY, JOB_TITLE
```
Comment (Evgenikaall) [inline homework-selects/sql_questions.sql:3]: For reserved keywords use uppercase, for names use lowercase. 
```
SELECT first_name
FROM employees

```

File: homework-selects/sql_questions.sql [hunk 3]
```diff
@@ -1,42 +1,197 @@
...
+FROM EMPLOYEES e
+         JOIN DEPARTMENTS d USING (DEPARTMENT_ID)
+         JOIN LOCATIONS l USING (LOCATION_ID);
+
 -- 3. the first name, last name, department number and department name, for all employees for departments 80 or 40.
+SELECT e.FIRST_NAME, e.LAST_NAME, d.DEPARTMENT_ID, d.DEPARTMENT_NAME
+FROM EMPLOYEES e
+         JOIN DEPARTMENTS d ON e.DEPARTMENT_ID = d.DEPARTMENT_ID
+WHERE e.DEPARTMENT_ID = 80
+   OR e.DEPARTMENT_ID = 40;
```
Comment (Evgenikaall) [inline homework-selects/sql_questions.sql:18]: also you can use here IN
`WHERE smthn IN (value1, value2)`

File: homework-selects/sql_questions.sql [hunk 4]
```diff
@@ -1,42 +1,197 @@
...
+FROM EMPLOYEES e1
+         JOIN EMPLOYEES e2 ON e1.MANAGER_ID = e2.EMPLOYEE_ID;
+
 -- 7. the first name of all employees and the first name of their manager including those who does not working under any manager.
+SELECT e1.FIRST_NAME AS EMPLOYEE_NAME, e2.FIRST_NAME AS MANAGER_NAME
+FROM EMPLOYEES e1
+         LEFT JOIN EMPLOYEES e2 ON e1.MANAGER_ID = e2.EMPLOYEE_ID;
+
 -- 8. the details of employees who manage a department.
+SELECT e.*
```
Comment (Evgenikaall) [inline homework-selects/sql_questions.sql:43]: Try to avoid usage of wildcards

File: homework-selects/sql_questions.sql [hunk 5]
```diff
@@ -1,42 +1,197 @@
...
+
 --20. all the information of the employees whose salary is within the range of smallest salary and 2500.
+SELECT *
+FROM EMPLOYEES
+WHERE SALARY BETWEEN (SELECT MIN(SALARY) FROM EMPLOYEES) AND 2500;
+
 --21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT e.*
+FROM EMPLOYEES e
+WHERE e.DEPARTMENT_ID NOT IN (SELECT DISTINCT DEPARTMENT_ID
```
Comment (Evgenikaall) [inline homework-selects/sql_questions.sql:121]: Try to fix formatting here

File: src/main/resources/db/migration/V20191004041103__create_table_projects.sql [hunk 6]
```diff
@@ -0,0 +1,19 @@
+CREATE TABLE PROJECTS
```
Comment (Evgenikaall) [inline src/main/resources/db/migration/V20191004041103__create_table_projects.sql:1]: Will be better to separate this declaration in 2 files
one for projects, one for project_empoyees

File: src/main/resources/db/migration/V20191004041104__create_trigger_1.sql [hunk 7]
```diff
@@ -0,0 +1,20 @@
+ALTER TABLE LOCATIONS
```
Comment (Evgenikaall) [inline src/main/resources/db/migration/V20191004041104__create_trigger_1.sql:1]: Please move alter table and comment in different file.

File: homework-selects/sql_questions.sql (hunk 1 above)
Comment (dani3lz) [reply homework-selects/sql_questions.sql]: Done

File: homework-selects/sql_questions.sql (hunk 2 above)
Comment (dani3lz) [reply homework-selects/sql_questions.sql]: Done

File: homework-selects/sql_questions.sql (hunk 3 above)
Comment (dani3lz) [reply homework-selects/sql_questions.sql]: Done

File: homework-selects/sql_questions.sql (hunk 4 above)
Comment (dani3lz) [reply homework-selects/sql_questions.sql:42]: Done

File: homework-selects/sql_questions.sql (hunk 5 above)
Comment (dani3lz) [reply homework-selects/sql_questions.sql]: Done

File: src/main/resources/db/migration/V20191004041103__create_table_projects.sql (hunk 6 above)
Comment (dani3lz) [reply src/main/resources/db/migration/V20191004041103__create_table_projects.sql:1]: Done

File: src/main/resources/db/migration/V20191004041104__create_trigger_1.sql (hunk 7 above)
Comment (dani3lz) [reply src/main/resources/db/migration/V20191004041104__create_trigger_1.sql]: Done

**Pull request merged**